pkg/controller/volume/persistentvolume/index.go

```diff
@@ -139,6 +141,10 @@ func (pvIndex *persistentVolumeOrderedIndex) findByClaim(claim *v1.PersistentVol
 			if v1helper.GetPersistentVolumeClass(volume) != requestedClass {
 				continue
 			}
+			pvVolumeMode := v1helper.GetPersistentVolumeVolumeMode(volume)
```

Comment:
The volume mode check also needs to be added [above](https://github.com/kubernetes/kubernetes/pull/53385/files#diff-161cd80c8bdf5f8596bca3bc00de37ebR124) for the prebound PV case.

Reply:
ack